addons/mail/static/src/js/views/activity/activity_renderer.js

```diff
@@ -51,25 +57,48 @@ var ActivityRenderer = AbstractRenderer.extend({
     },
     /**
      * @private
-     * @returns {jQueryElement} a jquery element <thead>
+     * @returns {jQueryElement} a <tfoot> element
      */
-    _renderHeader: function () {
-        var $tr = $('<tr>')
-                .append($('<th>')) //empty cell for name
-                .append(_.map(this.state.data.activity_types, this._renderHeaderCell.bind(this)));
-        return $('<thead>').append($tr);
+    _renderFooter: function () {
+        return QWeb.render('mail.ActivityViewFooter', {name: _.str.sprintf(_t('Add %s'), this.arch.attrs.string)});
     },
     /**
      * @private
-     * @param {Object} activity_type
-     * @returns {jQueryElement} a <th> element
+     * @returns {jQueryElement} a jquery element <thead>
      */
-    _renderHeaderCell: function (activity_type) {
-        return QWeb.render('mail.ActivityViewHeaderCell', {
-            id: activity_type[0],
-            name: activity_type[1],
-            template_list: activity_type[2] || [],
+    _renderHeader: function () {
+        var self = this;
+        var activityTypeIDs = _.unique(_.flatten(_.map(self.state.data.grouped_activities, function (item) {
+            return _.map(_.keys(item), Number);
+        })));
+        var $thead = $(QWeb.render('mail.ActivityViewHeader', {
+            types: this.state.data.activity_types,
+            activityTypeIDs: activityTypeIDs,
+        }));
+        _.each($thead.find('th.o_progress_bar'), function (th) {
+            var type = $(th).data('activity-type-id');
+            var counts = {planned: 0, today: 0, overdue: 0};
+            _.each(self.state.data.grouped_activities, function (act) {
+                if (_.contains(_.keys(act), type.toString())) {
+                    counts[act[type].state] += 1;
+                }
+            });
+
+            var progressBar = new ActivityColumnProgressBar(this,
+                {columnID: type, progressBarStates: {}},
+                {
+                    count: _.reduce(_.values(counts), function (x, y) { return x + y; }),
+                    progressBarValues: {
+                        field: 'activity_state',
+                        colors: {planned: 'success', today: 'warning', overdue: 'danger'},
+                        counts: counts,
+                    }
+                });
+            progressBar.appendTo($(th));
+            progressBar.on('tweak_activity_records', self, self._onTweakActivityRecords.bind(self));
+            progressBar.$counter.remove();
```

Comment:
This is not the responsibility of the activity renderer to remove `$counter` from `ActivityColumnProgressBar`.